hdl/XilinxBuildScript.tmpl

```diff
@@ -1,15 +1,39 @@
 #!/bin/bash
 set -eu -o pipefail
 
+WORK=`dirname {{ .Out }}`
+{{ $outOfContext := .OutOfContext }}
+
 cat > {{ .Out }} <<EOF
 {{range .BoardFiles}}
 set_param board.repoPaths [lappend board.repoPaths "{{ . }}"]
 {{end}}
 
 set_part "{{ .PartName }}"
-set_property "board_part"         "{{ .BoardName}}"       [current_project]
 set_property "target_language"    "Verilog"        [current_project]
 
+{{ if not .OutOfContext }}
+	set_property "board_part"         "{{ .BoardName}}"       [current_project]
+{{ end }}
+
+proc generate_reports {report_dir} {
+	# Timing
+	report_timing -from [all_registers -output_pins] -to [all_registers -data_pins] -max_paths 50 > \$report_dir/datapath_tss.rpt
```

Comment:
I'd incorporate the top-level name in the filenames of these reports and add them to the dbt rule's list of outputs.